docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,264 @@
+}
+function LoginButton(props) {
+  return <button onClick={props.click}>Login</button>
+}
+class LoginControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {loggedIn: true}
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
+  }
+  login() {
+    this.setState({loggedIn: true,});
+  }
+  logout() {
+    this.setState({loggedIn: false});
+  }
+  render() {
+    var loginControlButton = <LoginButton click={this.login} />;
+    if (this.state.loggedIn) {
+      loginControlButton = <LogoutButton click={this.logout} />;
+    }
+    return (
+      <div>
+        {loginControlButton}
+      </div>
+    );
+  }
+}
+
+```
+
+<a target="_blank" href="https://codepen.io/ericnakagawa/pen/Egpdrz?editors=0010">Try it out on Codepen.</a>
+
+
+### Prevent Component Rendering
+
+In some cases, you will not want a component to render. To prevent components from rendering, return `null` or `false` from the `render()` function.
+
+In the example below, a `<Page />` component renders a child `<WarningBanner />` element and a button. The button calls `this.toggleWarning` to switch the value of `this.state.showWarning`. If the `this.state.showWarning` is false, then nothing will be rendered.
+
+The highlighted lines below show where the variable `warningBanner` is set to a default value of null and then where it is rendered in `return()`.
+
+```javascript{17,23}
+function WarningBanner(props) {
+  return <div className="warning">Warning!</div>
+}
+class Page extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {showWarning: true, toggleText: "hide"}
+    this.toggleWarning = this.toggleWarning.bind(this);
+  }
+  toggleWarning() {
+    this.setState({
+      showWarning: !this.state.showWarning,
+      toggleText: !this.state.showWarning?"hide":"show"
+    });
+  }
+  render() {
+    let warningBanner = null;
+    if (this.state.showWarning) {
+      warningBanner = <WarningBanner />;
+    }
+    return (
+      <div>
+        {warningBanner}
+        <button onClick={this.toggleWarning}>{this.state.toggleText}</button>
+      </div>
+    );
+  }
+}
+```
+
+<a target="_blank" href="https://codepen.io/ericnakagawa/pen/ozragV?editors=0011#0">Try it out on Codepen.</a>
+
+### Rendering Multiple Components Using {}
+
+You can build collections of components and include them in <a target="_blank" href="/react/docs/introducing-jsx.html">JSX</a> with curly braces `{}` just as you would embed values with Javascript.
+
+In the example below we take an array of data `numbers`. Next, we add multiple `<Number />` components to the array.
+
+Then, in the `render()` function we render the entire `numbers` array by wrapping in curly braces `{numbers}`.
+
+```javascript
+function Number(props) {
+  return (
+    <li>{this.props.number}</li>
+  );
+}
+
+function Number(props) {
+  let numbers = [];
+  numbers.push(<Number value="1" />)
```

Comment:
This example will print React key warnings. It also gives a false impression that you would often write render this way (in practice I almost never saw push() usage in render).

Another confusing thing about it is it seems like a "special case" because it is separated from other sections but it is showing exact same technique as we used before: embedding arrays of elements.

Instead of a separate section I would make it a first step in the "loops" section.
First loops should show

```
const numbers = ...
const doubled = ...
```

The it should show 

```
const numbers = ...
const listItems = numbers.map(...)
ReactDOM.render(<ul>{listItems}</ul>, ...)
```

Then it should show component:

```
function NumberList(props) {
  const numbers = props.numbers
  const listItems = numbers.map(...)
  return <ul>{listItems}</ul>
}

const numbers = ...
ReactDOM.render(<NumberList numbers={...}, ...)
```

And only then show the "inline" notation

```
function NumberList(props) {
  const numbers = props.numbers
  return (
    <ul>
      {numbers.map(...)}
    </ul>
  )
}
```

This way we do exactly single step every time, and it's easy to keep track of what's changing and at the same time understand that it's just JavaScript.

As final step you could extract a Number component and use that to demonstrate key stays inside map.